tests/Integration/Presentation/Presenters/ConfirmSubscriptionHtmlPresenterTest.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+
+namespace WMDE\Fundraising\Frontend\Tests\Integration\Presentation\Presenters;
+
+use WMDE\Fundraising\Frontend\Presentation\Presenters\ConfirmSubscriptionHtmlPresenter;
+use WMDE\Fundraising\Frontend\ResponseModel\ValidationResponse;
+use WMDE\Fundraising\Frontend\TwigTemplate;
+use WMDE\Fundraising\Frontend\Validation\ConstraintViolation;
+
+/**
+ * @covers WMDE\Fundraising\Frontend\Presentation\Presenters\ConfirmSubscriptionHtmlPresenter
+ *
+ * @license GNU GPL v2+
+ * @author Gabriel Birke < gabriel.birke@wikimedia.de >
+ */
+class ConfirmSubscriptionHtmlPresenterTest extends \PHPUnit_Framework_TestCase {
+
+	public function testGivenSuccessResponse_templateIsRenderedWithoutMessages() {
+		$twig = $this->getMockBuilder( TwigTemplate::class )->disableOriginalConstructor()->getMock();
+		$twig->expects( $this->once() )
+			->method( 'render' )
+			->with( [] );
+		$presenter = new ConfirmSubscriptionHtmlPresenter( $twig );
+		$presenter->present( ValidationResponse::newSuccessResponse() );
+	}
```

Comment:
The way you wrote those tests, they are unit tests no?

What I did so far is use real twig and see certain HTML is present in the output. Not saying that approach is better, just pointing out a potential interesting difference.